shared/js/participantStore.js

```diff
@@ -39,17 +64,47 @@ loop.store.ParticipantStore = function() {
       return this._storeState.participants.size > 0;
     },
 
+    /**
+     * Handle LeaveRoom action by updating the current user's presence.
+     */
+    leaveRoom() {
+      this._updatePresence(false);
+    },
+
+    /**
+     * Handle SetOwnDisplayName action by saving the the current user's name.
+     */
+    setOwnDisplayName({ displayName }) {
+      this._currentUserObject.participantName = displayName;
+    },
+
     updatedParticipant(actionData) {
+      if (actionData.userId === this._currentUserId) {
+        if (this._currentUserObject) {
+          this._currentUserObject.participantName = actionData.participantName;
+        }
+        return;
+      }
+
       this._updateParticipantData(actionData.userId, {
         participantName: actionData.participantName
       });
     },
 
     updatedPresence(actionData) {
+      if (actionData.userId === this._currentUserId) {
+        return;
+      }
+
       this._updateParticipantData(actionData.userId, {
         isHere: actionData.isHere,
         localPingTime: Date.now() - actionData.pingedAgo
       });
+
+      // We just got some presence that we might need to expire later.
+      if (this._expireTimer === null) {
+        this._expireTimer = setTimeout(() => this._expirePresence(), PING_TIME);
```

Comment:
Added checks for timer variable being `null` or not as well as ticking the clock to make sure `_expirePresence` is not called yet or did.